doc/src/tutorials/physics/duffing-example.rst

```diff
@@ -0,0 +1,183 @@
+   >>> Pendulum = Block.locatenew('Pendulum', l * B.y)
+
+   >>> O.set_vel(N, 0)
+   >>> Block.set_vel(N, u1 * N.y)
+   >>> Pendulum.v2pt_theory(Block, N, B)
+   u1(t)*N.y - l*u2(t)*B.x
+
+   >>> ParBlock = me.Particle('ParBlock', Block, M)
+   >>> ParPendulum = me.Particle('ParPendulum', Pendulum, m)
+
+Define Force, Energy
+====================
+
+We obtain the Duffing force using the `DuffingSpring` actuator from ``sympy.physics.mechanics.actuator``.
+This force will be used to calculate the potential energy of the Duffing Oscillator block.
+
+.. plot::
+   :format: doctest
+   :include-source: True
+   :context: close-figs
+   :nofigs:
+
+   >>> # Define the Duffing spring force
+   >>> pathway = LinearPathway(O, Block)
+   >>> duffing_spring = DuffingSpring(k1, k2, pathway, 0)
+   >>> duffing_force = -duffing_spring.force
+   >>> duffing_force
+   k1*sqrt(q1(t)**2) + k2*(q1(t)**2)**(3/2)
+
+.. plot::
+   :format: doctest
+   :include-source: True
+   :context: close-figs
+   :nofigs:
+
+   >>> # Define Rayleigh dissipation
+   >>> D = (1/2) * c1 * q1d**2
+
+In relation to Lagrange's Method, we derive both the kinetic and potential energies of the system.
+
+.. plot::
+   :format: doctest
+   :include-source: True
+   :context: close-figs
+   :nofigs:
+
+   >>> # Kinetic Energy
+   >>> KBlock = (1/2) * M * Block.vel(N).dot(Block.vel(N))
+   >>> KPendulum = (1/2) * m * Pendulum.vel(N).dot(Pendulum.vel(N))
+   >>> Kinetic = KBlock + KPendulum
+   >>> Kinetic
+   0.5*M*u1(t)**2 + 0.5*m*(l**2*u2(t)**2 - 2*l*u1(t)*u2(t)*sin(q2(t)) + u1(t)**2)
+
+.. plot::
+   :format: doctest
+   :include-source: True
+   :context: close-figs
+   :nofigs:
+
+   >>> # Potential Energy
+   >>> PBlock = sm.integrate(duffing_force, q1)
+   >>> PPendulum = m * g * (l - l/(sm.sqrt(1+q2**2/q1**2)))
+   >>> Potential = PBlock + PPendulum
+   >>> Potential
+   g*m*(l - l/sqrt(1 + q2(t)**2/q1(t)**2)) + k1*sqrt(q1(t)**2)*q1(t)/2 + k2*(q1(t)**2)**(3/2)*q1(t)/4
+
+Lagrange's Method
+=================
+
+With the problem setup, the Lagrangian can be calculated, and the equations of motion formed.
+
+.. plot::
+   :format: doctest
+   :include-source: True
+   :context: close-figs
+   :nofigs:
+
+   >>> L = Kinetic - Potential
+   >>> L
+   0.5*M*u1(t)**2 - g*m*(l - l/sqrt(1 + q2(t)**2/q1(t)**2)) - k1*sqrt(q1(t)**2)*q1(t)/2 - k2*(q1(t)**2)**(3/2)*q1(t)/4 + 0.5*m*(l**2*u2(t)**2 - 2*l*u1(t)*u2(t)*sin(q2(t)) + u1(t)**2)
+
+   >>> FL = [(Block, duffing_force * N.y + D * N.y), (Pendulum, - m * g * N.y)] # [(point, the force acting on the particle)]
+   >>> LM = LagrangesMethod(L, [q1, q2], forcelist = FL, frame = N)
+   >>> LM.form_lagranges_equations()
+   Matrix([
+    [-g*l*m*q2(t)**2/((1 + q2(t)**2/q1(t)**2)**(3/2)*q1(t)**3) + k1*sqrt(q1(t)**2) + k2*(q1(t)**2)**(3/2)],
+    [              g*l*m*q2(t)/((1 + q2(t)**2/q1(t)**2)**(3/2)*q1(t)**2) + 1.0*l*m*u1(t)*u2(t)*cos(q2(t))]])
+
+References
+==========
+
+P. Brzeskia, P. Perlikowskia, S. Yanchukb, T. Kapitaniaka,
```

Comment:
Use the restructuredtext citation format: https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#citations